### PR TITLE
Update GO to 1.20

### PR DESCRIPTION
**Pull request by baoy1**
# Description
Update Go to 1.20

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/658|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have inspected the Grafana dashboards to verify the data is displayed properly
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] make check test
```
❯ make check test
./scripts/check.sh ./cmd/... ./internal/...
=== Checking format...
=== Finished
=== Vetting...
=== Finished
=== Linting...
=== Finished
=== Running gosec...
=== Finished
go test -count=1 -cover -race -timeout 30s -short ./...
?       github.com/dell/karavi-topology/cmd/topology    [no test files]
?       github.com/dell/karavi-topology/internal/entrypoint/mocks       [no test files]
ok      github.com/dell/karavi-topology/internal/entrypoint     0.031s  coverage: 100.0% of statements
?       github.com/dell/karavi-topology/internal/k8s/mocks      [no test files]
?       github.com/dell/karavi-topology/internal/service/mocks  [no test files]
?       github.com/dell/karavi-topology/internal/tracers        [no test files]
ok      github.com/dell/karavi-topology/internal/k8s    0.065s  coverage: 95.2% of statements
ok      github.com/dell/karavi-topology/internal/service        0.075s  coverage: 91.0% of statements
```

# Manual inspection of the GUI
I have verified that the dashboards show the data properly while generating I/O and storage resources

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by baoy1** [comment]
run e2e test

**Comment by baoy1** [comment]
run e2e test

File: internal/service/service.go [hunk 1]
```diff
@@ -96,7 +96,12 @@ func (s *Service) Run() error {
...
 		return fmt.Errorf("failed to listen on tcp port %d", s.Port)
 	}
-	defer ln.Close()
+	defer func(ln net.Listener) {
+		err := ln.Close()
+		if err != nil {
+			s.Logger.WithError(err).Error("failed to close listener")
+		}
+	}(ln)
 	tlsListener := tls.NewListener(ln, config)
```
Comment (forrestxia) [inline internal/service/service.go:105]: @baoy1 is this change to fix some golang version upgrade related issue? What if not changing it?

Comment (baoy1) [reply internal/service/service.go:105]: That fixes gosec error:
```
 G307 (CWE-703): Deferring unsafe method "Close" on type "net.Listener" (Confidence: HIGH, Severity: MEDIUM)
```

**Review by forrestxia — APPROVED** [review]

File: internal/service/service.go (hunk 1 above)
Comment (forrestxia) [reply internal/service/service.go:105]: got it. thanks for explaining it.

**Review by P-Cao — APPROVED** [review]

**Review by taohe1012 — APPROVED** [review]

**Pull request merged**